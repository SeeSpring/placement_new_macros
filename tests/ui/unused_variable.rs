use placement_new_macros::init;
use std::mem::MaybeUninit;

#[deny(unused_variables)]
fn main() {
    struct T {
        t: u32,
    }
    let mut v = MaybeUninit::<T>::uninit();
    init!(v, T { t: 0 }, f1 = 0);
}
