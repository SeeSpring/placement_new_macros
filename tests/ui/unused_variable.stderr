error: unused variable: `f1`
  --> $DIR/unused_variable.rs:10:26
   |
10 |     init!(v, T { t: 0 }, f1 = 0);
   |                          ^^ help: if this is intentional, prefix it with an underscore: `_f1`
   |
note: the lint level is defined here
  --> $DIR/unused_variable.rs:4:8
   |
4  | #[deny(unused_variables)]
   |        ^^^^^^^^^^^^^^^^
